src/lib/Libtpp/tpp_util.c

```diff
@@ -2332,61 +2330,65 @@ tpp_encrypt_pkt(conn_auth_t *authdata, tpp_packet_t *pkt)
 {
 	void *data_out = NULL;
 	size_t len_out = 0;
-	tpp_encrypt_hdr_t ehdr;
-	int totlen;
-	tpp_chunk_t *chunk;
-	tpp_chunk_t *first_chunk;
-
-	pkt->totlen = 0;
-	for (chunk = GET_NEXT(pkt->chunks); chunk; chunk = GET_NEXT(chunk->chunk_link)) {
-		if (authdata->encryptdef->encrypt_data(authdata->encryptctx, (void *) chunk->data, (size_t) chunk->len, &data_out, &len_out) != 0) {
-			return -1;
-		}
-
-		if (chunk->len > 0 && len_out <= 0) {
-			tpp_log(LOG_CRIT, __func__, "invalid encrypted data len: %d, pktlen: %d", (int) len_out, chunk->len);
-			return -1;
-		}
+	tpp_encrypt_hdr_t *ehdr;
+	int totlen = pkt->totlen;
+	tpp_chunk_t *chunk, *next;
+	tpp_auth_pkt_hdr_t *data = (tpp_auth_pkt_hdr_t *)(((tpp_chunk_t *)(GET_NEXT(pkt->chunks)))->data);
+	unsigned char type = data->type;
+	void *buf = NULL;
+	char *p;
 
-		/* now replace the data with encrypted buffer */
-		free(chunk->data);
-		chunk->data = data_out;
-		chunk->len = len_out;
-		chunk->pos = chunk->data;
+	if (type == TPP_AUTH_CTX && data->for_encrypt == FOR_ENCRYPT)
+		return 0;
 
-		pkt->totlen += len_out;
+	buf = malloc(totlen);
```

Comment:
oh but we will again malloc stuff. i think we should avoid this

Reply:
yes agree but I couldn't find any better way unless we somehow make chunks on receiver side instead full buffer... will chat offline...